src/object.c

```diff
@@ -532,6 +533,12 @@ void freeListObject(robj *o) {
 void freeSetObject(robj *o) {
     switch (o->encoding) {
     case OBJ_ENCODING_HT:
+#ifdef REDIS_TEST
+        dictEmpty((dict *) o->ptr, NULL);
+        size_t *alloc_size = (size_t *)dictMetadata((dict *) o->ptr);
+        UNUSED(alloc_size);
+        debugAssert(*alloc_size == 0);
```

Comment:
since it's already in REDIS_TEST, why no use serverAssert() instead of debugAssert().